### PR TITLE
Backend data

### PR DESCRIPTION
**Pull request by typeholes**
Moves the data and processing for traces to the extension for better performance.

Also includes the ability to create a mock of the extension panel for ease of development of the UI

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -6,5 +6,6 @@
   "eslint.lintTask.enable": false,
   "[typescript]": {
     "editor.defaultFormatter": "dbaeumer.vscode-eslint"
-  }
+  },
+  "tsperf.tracer.traceCmd": "/home/hw/projects/TypeScript/bin/tsc --noEmit --generateTrace ${traceDir}"
```
Comment (danielroe) [inline .vscode/settings.json]: this should probably not be committed

**Review by danielroe — COMMENTED** [review]
exciting!

I can't yet figure out how to run the trace though - what am I missing?

https://github.com/tsperf/tracer/assets/28706372/d39d04b0-9178-4127-a3b2-4a591484dadb

**Comment by typeholes** [comment]
You can run the trace by clicking on the status bar item or running the "Tracer: tsc trace" command

**Comment by typeholes** [comment]
fixed rare off by one in the goto position in trace and now streaming the tree start nodes in case there are enough to cause lag when sending all at once.

@ssalbdivad pointed out that overlapping error spans were confusing which highlighted that full spans didn't give the user any indication of what positions were in the trace, so I changed to single character spans for the trace diagnostics

**Comment by typeholes** [comment]
Ah you did run the command.  If it fails, the error output is shown in the terminal.   I set the default trace command to use npx, but that obviously doesn't work for all project.  here it should be pnpm.  Maybe I should just default to nothing and prompt the user on first trace run? 
image

**Comment by typeholes** [comment]
fixes #22

**Pull request merged**